components/public-api/gitpod/v1/user.proto

```diff
@@ -0,0 +1,212 @@
+      // +optional
+      string last_updated_details_nudge = 1;
+      // when was the last time the user has accepted our privacy policy
+      //
+      // +optional
+      string accepted_privacy_policy_date = 2;
+      // the user's company name
+      //
+      // +optional
+      string company_name = 3;
+      // the user's email
+      //
+      // +optional
+      string email_address = 4;
+      // type of role user has in their job
+      //
+      // +optional
+      string job_role = 5;
+      // freeform entry for job role user works in (when jobRole is "other")
+      //
+      // +optional
+      string job_role_other = 6;
+      // Reasons user is exploring Gitpod when they signed up
+      //
+      // +optional
+      repeated string exploration_reasons = 7;
+      // what user hopes to accomplish when they signed up
+      //
+      // +optional
+      repeated string signup_goals = 8;
+      // freeform entry for signup goals (when signupGoals is "other")
+      //
+      // +optional
+      string signup_goals_other = 10;
+      // Set after a user completes the onboarding flow
+      //
+      // +optional
+      string onboarded_timestamp = 11;
+      // Onboarding question about a user's company size
+      //
+      // +optional
+      string company_size = 12;
+    }
+
+    // remembered workspace auto start options
+    //
+    // +optional
+    repeated WorkspaceAutostartOption workspace_autostart_options = 9;
+    message WorkspaceAutostartOption {
+      string clone_url = 1;
+      string organization_id = 2;
+      string workspace_class = 3;
+      // +optional
+      IDESettings ide_settings = 4;
+      // +optional
+      string region = 5;
+    }
+  }
+
+  // usage_attribution_id specifies the organization or account to which all of the
+  // user's workspace usage is attributed to, e.g. for billing purposes
+  string usage_attribution_id = 11;
+
+  // per-user settings to control workspace behavior
+  repeated UserFeatureFlag workspace_feature_flags = 12;
+  enum UserFeatureFlag {
+    USER_FEATURE_FLAG_UNSPECIFIED = 0;
+    USER_FEATURE_FLAG_FULL_WORKSPACE_BACKUP = 1;
+    USER_FEATURE_FLAG_WORKSPACE_CLASS_LIMITING = 2;
+    USER_FEATURE_FLAG_WORKSPACE_CONNECTION_LIMITING = 3;
+    USER_FEATURE_FLAG_WORKSPACE_PSI = 4;
+  }
+
+  // specifies the permissions and/or roles the user has
+  repeated RoleOrPermission roles_or_permissions = 13;
+  enum RoleOrPermission {
+    ROLE_OR_PERMISSION_UNSPECIFIED = 0;
+    ROLE_OR_PERMISSION_DEVOPS = 1;
+    ROLE_OR_PERMISSION_VIEWER = 2;
+    ROLE_OR_PERMISSION_ADMIN = 3;
+    ROLE_OR_PERMISSION_DEVELOPER = 4;
+    ROLE_OR_PERMISSION_REGISTRY_ACCESS = 5;
+    ROLE_OR_PERMISSION_ADMIN_PERMISSIONS = 6;
+    ROLE_OR_PERMISSION_ADMIN_USERS = 7;
+    ROLE_OR_PERMISSION_ADMIN_WORKSPACE_CONTENT = 8;
+    ROLE_OR_PERMISSION_ADMIN_WORKSPACE = 9;
+    ROLE_OR_PERMISSION_ADMIN_PROJECTS = 10;
+    ROLE_OR_PERMISSION_NEW_WORKSPACE_CLUSTER = 11;
+  }
+}
+
+message Identity {
```

Comment:
Do we use it on the dashboard? 

It would be great try to change the dashboard code and see what we need or not, without any implementation yet?

Reply:
> Do we use it on the dashboard?

Yes.

This is what I did: search for attributes and see for references in dashboard. If not used there, then not include in here.